### PR TITLE
[expo-notifications] Split compound emitter

### PR DESCRIPTION
**Pull request by sjchmiela**
# Why

Follow up to https://github.com/expo/expo/pull/6773#discussion_r377340520.

# How

Split `addNotificationListener` into three: `addNotificationReceivedListener`, `addNotificationsDroppedListener`, `addNotificationResponseReceivedListener`.

# Test Plan

Added a test case to `test-suite`, verifying that `notificationReceivedListener` is called when a notification is received.

### CODE REVIEW

File: packages/expo-notifications/src/NotificationsEmitter.ts [hunk 1]
```diff
@@ -1,72 +1,38 @@
...
-export function removeNotificationSubscription(subscription: Subscription) {
-  if ('__subscriptions' in subscription) {
-    const compoundSubscription: CompoundSubscription = subscription;
-    for (let subscription of compoundSubscription.__subscriptions) {
-      emitter.removeSubscription(subscription);
-    }
-  } else {
-    emitter.removeSubscription(subscription);
-  }
+export function removeSubscription(subscription: Subscription) {
```
Comment (esamelson) [inline packages/expo-notifications/src/NotificationsEmitter.ts]: is there any difference between calling this and `subscription.remove()`? if not, this seems redundant.

**Review by esamelson — APPROVED** [review]

File: packages/expo-notifications/src/NotificationsEmitter.ts (hunk 1 above)
Comment (sjchmiela) [reply packages/expo-notifications/src/NotificationsEmitter.ts]: True, I agree! This is also what James pointed out in https://github.com/expo/expo/pull/6577#discussion_r367219901

> ~removeTokenSubscription~ removePushTokenSubscription
> (although, is this function needed if developers can just call `subscription.remove()`?)

, to which I responded with https://github.com/expo/expo/pull/6779#issuecomment-575152833

> I agree developers can just call `subscription.remove()`, but eg. `EventEmitter.ts` and other event emitters (like `MediaLibrary`) implement this method anyway.
> 
> Maybe if we decide to remove this method from all event emitters it should be in a separate PR?

which has been approved of. Do you think we should remove this method now, or should we remove all of them immediately, or postpone the change until… some day?

Comment (esamelson) [reply packages/expo-notifications/src/NotificationsEmitter.ts]: Gotcha. Sorry to make you justify this twice!! Let's leave it for now for consistency.

Comment (sjchmiela) [reply packages/expo-notifications/src/NotificationsEmitter.ts]: This only proves that the review process is consistently good and thorough! 😃

**Pull request merged**